src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -450,13 +429,11 @@ private sealed class ScriptEntryPoint : SynthesizedEntryPointSymbol
             //     var script = new Script();
             //     script.<Initialize>().GetAwaiter().GetResult();
             // }
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
             {
-                // CreateBody should only be called if no errors.
-                Debug.Assert((object)_getAwaiterMethod != null);
-                Debug.Assert((object)_getResultMethod != null);
-
                 var syntax = DummySyntax();
+                var compilation = _containingType.DeclaringCompilation;
+                var binder = compilation.GetBinder((CSharpSyntaxNode)_containingType.DeclaringSyntaxReferences[0].SyntaxTree.GetRoot());
```

Comment:
Ok, we should capture that in some comment. I do not think that simply checking the binder under debugger is good enough. We should have targeted tests that would verify that global usings are taken into account, but the non-global ones are not.